data-prepper-plugins/dynamodb-source/src/main/java/org/opensearch/dataprepper/plugins/source/dynamodb/converter/StreamRecordConverter.java

```diff
@@ -80,6 +80,7 @@ public void writeToBuffer(final AcknowledgementSet acknowledgementSet, List<Reco
         int eventCount = 0;
         for (Record record : records) {
             final long bytes = record.dynamodb().sizeBytes();
+            final Boolean userIdentity = record.userIdentity() != null && "dynamodb.amazonaws.com".equals(record.userIdentity().principalId());
```

Comment:
Also, I just checked the documentation it states that `Records[].userIdentity.type` will equal `"Service"`. Let's also check that condition.

Reply:
Added the extra check for `userIdentity.type`